internal/kibana/synthetics/schema.go

```diff
@@ -1,15 +1,299 @@
 package synthetics
 
 import (
+	"fmt"
 	"github.com/disaster37/go-kibana-rest/v8/kbapi"
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
 	"github.com/hashicorp/terraform-plugin-framework/types"
+	"strconv"
 )
 
 const (
 	MetadataPrefix = "_kibana_synthetics_"
 )
 
+type kibanaAPIRequest struct {
+	fields kbapi.MonitorFields
+	config kbapi.SyntheticsMonitorConfig
+}
+
+type tfStatusConfigV0 struct {
+	Enabled types.Bool `tfsdk:"enabled"`
+}
+
+type tfAlertConfigV0 struct {
+	Status *tfStatusConfigV0 `tfsdk:"status"`
+	TLS    *tfStatusConfigV0 `tfsdk:"tls"`
+}
+
+type tfHTTPMonitorFieldsV0 struct {
+	URL                   types.String   `tfsdk:"url"`
+	SslVerificationMode   types.String   `tfsdk:"ssl_verification_mode"`
+	SslSupportedProtocols []types.String `tfsdk:"ssl_supported_protocols"`
+	MaxRedirects          types.String   `tfsdk:"max_redirects"`
+	Mode                  types.String   `tfsdk:"mode"`
+	IPv4                  types.Bool     `tfsdk:"ipv4"`
+	IPv6                  types.Bool     `tfsdk:"ipv6"`
+	ProxyURL              types.String   `tfsdk:"proxy_url"`
+	// commented out due-to https://github.com/elastic/kibana/issues/189906
+	// it leads to terraform  Error: Provider produced inconsistent result after apply
+	//ProxyHeader           jsontypes.Normalized `tfsdk:"proxy_header"`
+	//Username              types.String         `tfsdk:"username"`
+	//Password              types.String         `tfsdk:"password"`
```

Comment:
These three in particular feel fairly important to support. IIUC these fields are just not present in a GET. IMO it would be preferable to support these attributes during creation only, than not supporting them at all. To be clear, I'm suggesting we let the user configure them, Terraform just won't detect drift from the desired state since the current values are not available during the read cycle.

Reply:
> to be clear, I'm suggesting we let the user configure them, Terraform just won't detect drift from the desired state since the current values are not available during the read cycle.

Terraform detects drift because PUT / POST returns this values back. Do you suggest to remove them from PUT / POST response mapping as well?

Reply:
> Terraform detects drift because PUT / POST returns this values back.

IIUC the current workflow is something like (assuming these fields still existed):
1. TF creates the monitor. We set state based on the response (including user/pass)
2. TF refresh reads the existing monitor. These fields aren't included in the response and so they're updated to be empty
3. TF apply notices empty != configured and causes and updates the monitor
4. Update sets the fields in state again (since they're returned in the response)
5. Loop back to 2.

Is that right, or is there something else about PUT/POST I'm missing here?

There's a couple of changes we can make here:
* Most (all?) of the other resources discard the PUT/POST responses tbh. Instead they get the state model via the read. In some cases there's technical reasons for that, in other cases it's mostly to have a consistent read path. 
* Instead of building an entirely new state model from the response (like [here](https://github.com/elastic/terraform-provider-elasticstack/pull/699/files#diff-1cfb31628c813a5448fbf4e10535e71f3233bcb07692b41eb7316f67c54613aaR42), we update the existing state, preserving the user/pass fields if they're unset in the response. 

WDYT?

Reply:
> Instead they get the state model via the read.

not sure, if that will work (I don't know how TF will handle that fields become empty). Currently, if `TF apply notices empty != configured` it fails the test. If I allow state be based on read, it would eventually set not-returned fields to empty, but they will stay configured on the server, because of [partial update](https://www.elastic.co/guide/en/kibana/current/update-monitor-api.html#_partial_updates) API behaviour. 

> we update the existing state, preserving the user/pass fields if they're unset in the response.

so merge. That might be a good idea, let me try, if I could implement it.